src/test/java/com/google/devtools/build/lib/analysis/LocationExpanderIntegrationTest.java

```diff
@@ -158,6 +162,10 @@ public void otherPathExternalExpansion() throws Exception {
         .matches("foo external/r/p/foo.txt bar");
     assertThat(expander.expand("foo $(rootpaths @r//p:foo) bar"))
         .matches("foo external/r/p/foo.txt bar");
+    assertThat(expander.expand("foo $(rlocationpath @r//p:foo) bar"))
+      .matches("foo r/p/foo.txt bar");
+    assertThat(expander.expand("foo $(rlocationpath @r//p:foo) bar"))
```

Comment:
Please add test for `$(rlocationpath @//p:foo)` and `$(rlocationpath @main_repo//p:foo)`, where `main_repo` matches main repo's name. 

Not necessarily here, please pick the best location for those.